pilot/pkg/security/authz/model/model.go

```diff
@@ -160,6 +164,52 @@ func (m *Model) MigrateTrustDomain(tdBundle trustdomain.Bundle) {
 	}
 }
 
+// Update rules for ambient based on listener name. Because the Waypoint Proxy receives
+// the destination as envoy://listener_name/, Envoy will not have a port to evaluate. This
+// will precalculate the RBAC rules with regard to destination port for any matches and replace
+// as necessary. For not matches, we add an `any: true` as it will always match. For value match,
+// we replace with a nil, which the generator replaces with an `any: true` if there are no other
+// rules to evaluate. We leave non-matching ports so they do not match any requests.
+func (m *Model) AmbientDestinationPortAdapations(listener_name string) {
```

Comment:
This is pretty complex. We have golden file tests (in Yossi's PR, I think specifically for Ambient mode if you want to copy); may be useful to include these.

Reply:
I am looking at those unit tests, I need to be able to pass a listener name.